### ISSUE TITLE
first_bn/FusedBatchNorm_mul_0, which is an input to the Add operator producing the output array kws_model/KWS_Model/tower_0/CNN_V1/Relu, is lacking min/max data, which is necessary for quantization. Either target a non-quantized output format, or change the input graph to contain min/max information, or pass --default_ranges_min= and --default_ranges_max= if you do not care about the accuracy of results.\nAborted (core dumped)"

### ISSUE THREAD

**Issue by hahadashi** [tensorflow/tensorflow#28000]
**System information**
-Centos
- TensorFlow 1.9:

**change the pb model to tflite *
my network include batch_normalization layer, 
when I change the the pb model to tflite , there is always return a error message.
```
first_bn/FusedBatchNorm_mul_0, which is an input to the Add operator producing the output array kws_model/KWS_Model/tower_0/CNN_V1/Relu, is lacking min/max data, which is necessary for quantization. Either target a non-quantized output format, or change the input graph to contain min/max information, or pass --default_ranges_min= and --default_ranges_max= if you do not care about the accuracy of results.\nAborted (core dumped)"
```

I tried  lots of methods, like add  converter.default_ranges_stats=(0, 6) or remove bn layers. but  I know its not the expected ways. hope other friends can give me some suggestions. thanks

**Comment by achandraa** [comment]
Thank you for reaching out to us. Can you try with the latest tensorflow version since tensorflow 1.9 seems to be a bit older version. Let us know if you still face the issue.

**Comment by hahadashi** [comment]
> Thank you for reaching out to us. Can you try with the latest tensorflow version since tensorflow 1.9 seems to be a bit older version. Let us know if you still face the issue.

i have tried the 1.12, but i still face the problem.

**Comment by hahadashi** [comment]
> Thank you for reaching out to us. Can you try with the latest tensorflow version since tensorflow 1.9 seems to be a bit older version. Let us know if you still face the issue.

thanks

**Comment by ruffles1** [comment]
I'm having the same issue with quantizing a model containing batch norm layers, which I've previously reported, and referenced here:

https://github.com/tensorflow/tensorflow/issues/25301

**Comment by hahadashi** [comment]
> I'm having the same issue with quantizing a model containing batch norm layers, which I've previously reported, and referenced here:
> 
> #25301

thanks

**Comment by achandraa** [comment]
@hahadashi : Were you able to resolve the issue ?

**Comment by hahadashi** [comment]
> @hahadashi : Were you able to resolve the issue ?

no, i have tried my best to find a correct method, but it didn't work. during the last days, i find lots of people meet the same problem (tf.layer.batch_normalization), but all can not resolve it. hope you can give some suggestions.

**Comment by achandraa** [comment]
@hahadashi : Did you get chance to try out solution provided in #21725 or #22528. Looks like we might have to add FakeQuantWithMinMaxVars. Let us know. Thanks!

**Comment by hahadashi** [comment]
> @hahadashi : Did you get chance to try out solution provided in #21725 or #22528. Looks like we might have to add FakeQuantWithMinMaxVars. Let us know. Thanks!
@achandraa  thanks.
in  #21725 or #22528, they give a solution that set "default range value". i can resolve my problem by this way, but we known this method may bring error range.

**Comment by ruffles1** [comment]
@achandraa Setting default ranges for ops with missing min/max can't possibly be an acceptable solution. 

The error message clearly states that it should only be used if accuracy isn't important (in what scenario would this be true other than for experimentation?):

`Either target a non-quantized output format, or change the input graph to contain min/max information, or pass --default_ranges_min= and --default_ranges_max= if you do not care about the accuracy of results.`

**Comment by cuongdv1** [comment]
I have same issue, were anyone able to resolve the issue ?
Thanks

**Comment by cuongdv1** [comment]
@jianlijianli 
Do you have any information for resolving this issue? please help
This issue occurs when I convert quantized model (using quantization aware training) to TFLite

**Comment by lrigazio** [comment]
same issue here on tf-1.15

**Comment by Shoshin23** [comment]
Same issue here. Has anyone managed to fix this? I'm getting a FusedBatchNormV3 error when converting my mobilenetv3 model to TFlite.

**Comment by ppershing** [comment]
Same problem with FusedBatchNormV3 - Add - Relu sequence when trying to do quantization-aware training using contrib.quantize

**Comment by dtlam26** [comment]
For fused_batch norm, you should turn off by set `fused=False`. By inspecting the contrib [https://github.com/tensorflow/tensorflow/blob/r1.15/tensorflow/contrib/quantize/python/quantize.py#L35](url)
The only way to remove this is to edit the model so that the activation listing in here will be at the last layer lead to quantize activation. You can check this stackoverflow, I have made it work on BatchNorm as an example [https://stackoverflow.com/questions/63681168/batch-normalization-quantize-tensorflow-1-x-does-not-have-minmax-information](url)

**Comment by mohantym** [comment]
Hi @hahadashi !
It seems you are using older versions(1.x versions) of Tensorflow. We recommend that you upgrade  your code base to 2.x  versions as many features and bug fixes has been done in newer versions and let us know if the issue still persists in newer versions. Thanks!
